packages/salesforcedx-slds-linter/src/server/index.ts

```diff
@@ -0,0 +1,207 @@
+// when the text document first opened or when its content has changed.
+documents.onDidChangeContent(change => {
+  validateTextDocument(change.document.getText(), change.document.uri, connection);
+});
+
+documents.onDidOpen(change => {
+  validateTextDocument(change.document.getText(), change.document.uri, connection);
+});
+
+// The settings have changed. Is send on server activation
+// as well.
+connection.onDidChangeConfiguration(change => {
+  // Revalidate any open text documents
+  documents.all().forEach(doc => validateTextDocument(doc.getText(), doc.uri, connection));
+});
+
+let activeDiagnostics: Diagnostic[] = [];
+
+export function validateTextDocument(textDocument: String, uri: string, conn: any): void {
+  activeDiagnostics = [];
+  const lines = textDocument.split(/\r?\n/g);
+  let problems = 0;
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i];
+    const found = line.match(/slds\S*--[A-Za-z0-9_-]+/g) || [];
+    for (const match of found) {
+      const index = line.search(match) || -1;
+
+      if (index >= 0) {
+        const foundStringLength = match.length;
+        const fixedString = match.replace('--', '_');
+        problems++;
+        const diagnostic = <Diagnostic>{
+          code: `0${fixedString}`,
+          severity: DiagnosticSeverity.Warning,
+          range: {
+            start: { line: i, character: index },
+            end: { line: i, character: index + foundStringLength }
+          },
+          message: nls.localize('deprecated_class_name', line.substr(
+            index,
+            foundStringLength
+          ), fixedString),
+          source: 'slds'
+        };
+        activeDiagnostics.push(diagnostic);
+      }
+    }
+  }
+
+  // Send the computed diagnostics to VSCode.
+  conn.sendDiagnostics({
+    uri: uri,
+    diagnostics: activeDiagnostics
+  });
+}
+
+connection.onCodeAction(params => {
+  const uri = params.textDocument.uri;
+  const diagnostics = params.context.diagnostics;
+  const result: Command[] = [];
+  const edits: TextEdit[] = [];
+  let code = '';
+
+  for (const diagnostic of diagnostics) {
+    const codeStr = <string>diagnostic.code;
+    code = codeStr[0];
+    const replacementStr = codeStr.slice(1);
+
+    switch (code) {
+      case '0': {
+        edits.push({
+          range: diagnostic.range,
+          newText: replacementStr
+        });
+
+        result.push(
+          Command.create(
+            nls.localize('fix_problem', diagnostic.message),
+            'sfdx.force.lightning.slds.fix.deprecated.class',
+            uri,
+            edits
+          )
+        );
+      }
+    }
+  }
+
+  sameCodeActions(result, uri, code);
+  allCodeActions(result, uri);
+
+  return result;
+});
+
+function allCodeActions(result: Command[], uri: string) {
+  const fixAllEdits: TextEdit[] = [];
+
+  if (activeDiagnostics.length > 1) {
+    for (const codeAction of activeDiagnostics) {
+      const codeStr = <string>codeAction.code;
```

Comment:
Cosmetics: Please use `codeAction.code as string` for casting. See http://acdcjunior.github.io/typescript-cast-object-to-other-type-or-instanceof.html for why we want to avoid using `<>`